nft/handler.go

```diff
@@ -64,6 +64,40 @@ func (g grpcHandler) MintNFT(ctx context.Context, request *nftpb.NFTMintRequest)
 	}, nil
 }
 
+// MintPaymentObligationNFT will be called from the client API to mint an NFT out of an invoice payment obligation
+func (g grpcHandler) MintPaymentObligationNFT(ctx context.Context, request *nftpb.NFTPaymentObligationRequest) (*nftpb.NFTMintResponse, error) {
+	apiLog.Infof("Received request to Mint a Payment Obligation NFT for invoice %s and deposit address %s", request.Identifier, request.DepositAddress)
+	ctxHeader, err := contextutil.Context(ctx, g.config)
+	if err != nil {
+		apiLog.Error(err)
+		return nil, err
+	}
+
+	// Get proof fields
+	proofFields, err := g.service.GetRequiredPaymentObligationProofFields(ctxHeader)
+	if err != nil {
+		return nil, centerrors.New(code.Unknown, err.Error())
+	}
+
+	cfg, err := g.config.GetConfig()
+	if err != nil {
+		return nil, centerrors.New(code.Unknown, err.Error())
+	}
+	poRegistry := cfg.GetContractAddress(config.PaymentObligation)
+
+	mintReq := &nftpb.NFTMintRequest{
+		Identifier:                request.Identifier,
+		DepositAddress:            request.DepositAddress,
+		RegistryAddress:           poRegistry.Hex(),
+		ProofFields:               proofFields,
+		GrantNftAccess:            true,
+		SubmitNftOwnerAccessProof: true,
```

Comment:
can you merge `GrantNftAccess` and `SubmitNftOwnerAccessProof` to `GrantNftAccess` alone? These two must always be set to true.

Reply:
I rather do that as a separate PR, as there is logic change, functional tests ...